test/gtest/uct/test_flush.cc

```diff
@@ -538,3 +539,255 @@ UCS_TEST_SKIP_COND_P(uct_flush_test, am_pending_flush_nb,
 }
 
 UCT_INSTANTIATE_TEST_CASE(uct_flush_test)
+
+struct uct_cancel_test : public uct_test {
+    struct side {
+        side(uct_cancel_test *test, side *peer) :
+            test(*test), peer(peer), clean_mem(false),
+            e(NULL), buf1k(NULL), buf8k(NULL), buf32(NULL) {}
+
+        uct_cancel_test &test;
+        struct side *peer;
+        bool clean_mem;
+        entity *e;
+        mapped_buffer *buf1k;
+        mapped_buffer *buf8k;
+        mapped_buffer *buf32;
+
+        void init() {
```

Comment:
- why not ctor/dtor?
- use private when possible

Reply:
to call it from SetUp context

Reply:
peer does not inherit from uct_test, we can construct it from SetUp